modules/src/core/ics02_client/context.rs

```diff
@@ -73,16 +74,24 @@ pub trait ClientReader {
         &self,
         height: Height,
         proof: Option<Vec<u8>>,
-    ) -> Result<AnyConsensusState, Error>;
+    ) -> Result<Self::ConsensusState, Error>;
 
     /// Returns a natural number, counting how many clients have been created thus far.
     /// The value of this counter should increase only via method `ClientKeeper::increase_client_counter`.
     fn client_counter(&self) -> Result<u64, Error>;
 }
 
 /// Defines the write-only part of ICS2 (client functions) context.
-pub trait ClientKeeper {
-    fn store_client_result(&mut self, handler_res: ClientResult) -> Result<(), Error> {
+pub trait ClientKeeper
+where
+    Self: Sized + Clone + Debug + Eq,
```

Comment:
why is Sized needed here?

Reply:
Yeah, it's not needed actually, I don't remember why I've added it here